### PR TITLE
fix: RHEL compatibility for systemd service ordering

### PR DESCRIPTION
**Pull request by ThrRip**
On RHEL-based Linux distros, `nss-lookup.target` is deprecated and remains `inactive (dead)` forever by default.
```console
$ systemctl status nss-lookup.target
○ nss-lookup.target - Host and Network Name Lookups
     Loaded: loaded (/usr/lib/systemd/system/nss-lookup.target; static)
     Active: inactive (dead)
       Docs: man:systemd.special(7)
```
This pull request replaces its usage in the service file with `network-online.target`, which is reached when the network is "up", and is also available on Debian-based distros. More information can be found at [https://systemd.io/NETWORK_ONLINE/](https://systemd.io/NETWORK_ONLINE/).

### CODE REVIEW

**Comment by layou233** [comment]
Thanks for your PR!

I searched for service files that included by popular projects, and I found sing-box and Xray are still using `nss-lookup.target`.

https://github.com/XTLS/Xray-install/blob/9af684cf00e129231e6861985f3c4b2ce67e2ffc/install-release.sh#L515

https://github.com/SagerNet/sing-box/blob/951d83cf9774e1ba785c90d36830d04c64942000/release/local/sing-box.service#L4

And in https://systemd.io/NETWORK_ONLINE/ they are also using `nss-lookup.target` in their FAQ.

Are they facing the same problem?

**Comment by ThrRip** [comment]
> Are they facing the same problem?

Sometimes. `nss-lookup.target` still works as expected on Debian-based Linux distros (tested on Ubuntu 22.04 LTS), but not on RHEL-based distros anymore (tested on AlmaLinux 8 & 9).

In reality, `network.target` is reached when the system **starts** to bring the network up, and `network-online.target` when it **finishes** to do so. This operation takes a few seconds, while `nss-lookup.target` is reached somewhere in between them, when the hostname resolver becomes available.

Due to `nss-lookup.target` is inactive (again, only on RHEL-based distros), `After=network.target nss-lookup.target` will just start the service right after `network.target` is reached. If no network requests that requires access to addresses **other than** `127.0.0.1`, `0.0.0.0`, or `::` are made in this short span, everything will be fine - and this is most of the cases.

So in conclusion, all these systemd service files, including the one in this project, they all work. Just might not be the most optimal ones, and make less sense.

**Comment by ThrRip** [comment]
I also asked Bing AI and Bard for this, and they're all giving answers that are different from each other's though. A comment is good to have.

> Although I don't think it's very likely that anyone would use this example service we provided LOL

You got the point.

**Comment by layou233** [comment]
Merged. Thanks again for your PR and your patience!
